### ISSUE TITLE
Multiple options with the same aria-label

### ISSUE THREAD

**Issue by francescovenica** [Hacker0x01/react-datepicker#4680]
**Describe the bug**

When using range and 2 month the left month contains a few day of the next month (1st,2nd,3rd, and other based on the month), this create issues when testing (and probably with screen reader) as doing something like:

```
await page.getByLabel(`Choose ${todayPlusMonthFormatted}`, { exact: true }).click();
```

will fail because it find 2 options.

Here a codesendbox

**Expected behavior**
The hidden option shouldn't have the aria-label or should be disabled

**Screenshots**

<img width="401" alt="Screenshot 2024-04-11 at 08 19 40" src="https://github.com/Hacker0x01/react-datepicker/assets/30143154/dabddd98-0ca8-4f08-9635-38b5b2911d3c">
